### DISCUSSION TITLE
Rewiring / Extending Conditional UI - An idea

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7442] [Ideas]
The premise of the Idea is to be able to set multiple settings on a single condition, which would work in parallel to the current mechanism. The proposed idea, is to allow the user to implement the following rule (pseudo code)

IF Device.Mobile === true 
1. SET Direction to Column
2. SET Padding to 4px
3. SET Text = {binding}

With the current mechanism, this would require 3 different rules, whilst the proposed way you could group them into one, and be able to better maintain an extensive list of rules.

**Comment by andz-bb** [comment]
Hey @poirazis, I like this one, thanks for the idea! I could see it being useful to group the settings using one condition together under the same condition in the UI. Is it just a visual change for clarity you are thinking of or can you imagine any additional functionality coming from this? Cheers

**Reply by poirazis** [reply]
hey @andz-bb , I was thinking that apart from the visual arrangement, the evaluation gain in speed, as the IF condition would be evaluated only once, and if true then loop the items/conditionals and set the parameters.
